src/template.py

```diff
@@ -8,17 +8,23 @@
 """
 
 import numpy as np
+import sys
 from .constants import TEMPLATE_DEFAULTS_PATH, QTYPE_DATA
-from .utils.file import load_json
+from .utils.file import load_json,validate_json
 from .utils.object import OVERRIDE_MERGER
 
 TEMPLATE_DEFAULTS = load_json(TEMPLATE_DEFAULTS_PATH)
 
 
-def open_template_with_defaults(template_path):
+def open_template_with_defaults(schema_path,template_path):
     user_template = load_json(template_path)
-    return OVERRIDE_MERGER.merge(TEMPLATE_DEFAULTS, user_template)
-
+    user_template = OVERRIDE_MERGER.merge(TEMPLATE_DEFAULTS, user_template)
+    is_valid, msg = validate_json(user_template,schema_path)
+    if is_valid:
```

Comment:
add new line above

Reply:
Added newline and other spacings in template.py